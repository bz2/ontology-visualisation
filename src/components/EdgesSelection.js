import { useState, useEffect } from 'react'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { Button } from 'primereact/button'
import actions from '../store/actions'
import EdgesSelectionDetails from './EdgesSelectionDetails'
import { SIDEBAR_VIEW_EDGES_SELECTION } from '../constants/views'
import { getEdgeUniqueId } from '../constants/functions'

const EdgesSelection = ({
  removeFromArray,
  objectPropertiesFromApi,
  selectedEdges,
  resetSelectedEdges
}) => {
  const { t } = useTranslation()

  const [isEdgeSelected, toggleEdgeSelected] = useState(false)
  const [edgeId, setEdgeId] = useState('')

  useEffect(() => () => resetSelectedEdges(), [])

  return (
    <>
      <div className="sidebar-main-title">
        {!isEdgeSelected
          ? t(SIDEBAR_VIEW_EDGES_SELECTION)
          : (
            <>
              <Button
                tooltip={t('goBack')}
                onClick={() => toggleEdgeSelected(false)}
                icon="pi pi-chevron-left"
                iconPos="left"
              />
              {`${t('edge')}: ${objectPropertiesFromApi[edgeId].rdfsLabel}`}
            </>
          )}
      </div>
      {!isEdgeSelected ? (
        <div className="edges-selection">
          {
            selectedEdges.length > 0
              ? selectedEdges.map((selectedEdge) => {
                const edgeUniqueId = getEdgeUniqueId(selectedEdge)

                const { rdfsLabel } = objectPropertiesFromApi[edgeUniqueId]

                return (
                  <div
                    className="edges-selection-row"
                    key={`selected-edge-row-${selectedEdge}`}
                  >
                    <div className="edges-selection-row-delete">
                      <Button
                        tooltip={`${t('removeEdge')}: ${rdfsLabel}`}
                        onClick={() => removeFromArray('selectedEdges', selectedEdge)}
                        icon="pi pi-times"
                      />
                    </div>

                    <div className="edges-selection-row-main">
                      <Button
                        tooltip={`${t('viewEdge')}: ${rdfsLabel}`}
                        onClick={() => {
                          setEdgeId(edgeUniqueId)
                          toggleEdgeSelected(true)
                        }}
                        label={rdfsLabel}
                        icon="pi pi-chevron-right"
                        iconPos="right"
                      />
                    </div>
                  </div>
                )
              }) : (
                <div className="edges-selection-message">
                  {t('selectEdgeFromGraph')}
                </div>
              )
          }
        </div>
      ) : (
        <EdgesSelectionDetails
          edgeId={edgeId}
        />
      )}
    </>
  )
}

EdgesSelection.propTypes = {
  resetSelectedEdges: PropTypes.func.isRequired,
  selectedEdges: PropTypes.arrayOf(PropTypes.string).isRequired,
  removeFromArray: PropTypes.func.isRequired,
  objectPropertiesFromApi: PropTypes.shape().isRequired,
}

const mapToProps = ({
  selectedEdges,
  objectPropertiesFromApi
}) => ({
  selectedEdges,
  objectPropertiesFromApi
})

export default connect(
  mapToProps,
  actions
)(EdgesSelection)
