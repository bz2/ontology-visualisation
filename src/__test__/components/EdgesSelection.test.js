import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EdgesSelection from '../../components/EdgesSelection'

const setup = () => {
  const props = {
    setStoreState: jest.fn(),
    selectedEdge: 'edge-123'
  }

  const component = shallow(<EdgesSelection {...props} />)

  return {
    component,
    props
  }
}

describe('EdgesSelection', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
