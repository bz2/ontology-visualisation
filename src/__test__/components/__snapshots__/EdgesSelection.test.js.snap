// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EdgesSelection should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "addNumber": [Function],
      "addOrAppendToObject": [Function],
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "addValueToSubkeyObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
      "toggleFromArrayInKey": [Function],
      "toggleFromSubArray": [Function],
    }
  }
  mapToProps={[Function]}
  selectedEdge="edge-123"
  setStoreState={[MockFunction]}
>
  <Component />
</Connect>
`;
